addons/purchase/models/purchase.py

```diff
@@ -433,6 +439,71 @@ def action_view_invoice(self):
         result['context']['default_ref'] = self.partner_ref
         return result
 
+    @api.model
+    def retrieve_dashboard(self):
+        """ This function returns the values to populate the custom dashboard in
+            the purchase order views.
+        """
+        result = {
+            'all_to_send': 0,
+            'all_waiting': 0,
+            'all_late': 0,
+            'my_to_send': 0,
+            'my_waiting': 0,
+            'my_late': 0,
+            'all_avg_order_value': 0,
+            'all_avg_days_to_purchase': 0,
+            'all_total_last_7_days': 0,
+            'all_sent_rfqs': 0,
+            'company_currency_symbol': self.env.company.currency_id.symbol
+        }
+
+        one_week_ago = fields.Datetime.to_string(fields.Datetime.now() - relativedelta(days=7))
+        # This query is brittle since it depends on the label values of a selection field
+        # not changing, but we don't have a direct time tracker of when a state changes
+        query = """SELECT COUNT(1)
+                   FROM mail_tracking_value v
+                   LEFT JOIN mail_message m ON (v.mail_message_id = m.id)
+                   JOIN purchase_order po ON (po.id = m.res_id)
+                   WHERE m.create_date >= '%s'
+                     AND m.model = 'purchase.order'
+                     AND m.message_type = 'notification'
+                     AND v.old_value_char = 'RFQ'
+                     AND v.new_value_char = 'RFQ Sent'
+                     AND po.company_id = %s;
+                """ % (one_week_ago, self.env.company.id)
```

Comment:
In a security point of view, It is highly recommended to **not** interpolate your string SQL query.
Check this future security documentation: https://github.com/odoo/odoo/pull/43463/commits/9ca31d9f0f0417018853b591c9cbd107edcc7b79#diff-9ac482b374467f6d41c3b4fb3b3b48cfR166